firewood/src/merkle.rs

```diff
@@ -277,7 +261,9 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
                                             result = Err(e);
                                         }
                                     }
-                                    NodeType::Branch(_) => unreachable!(),
+                                    NodeType::Branch(u) => {
```

Comment:
why is this possible now? A comment would be helpful

Reply:
Because branch nodes can split now